### ISSUE TITLE
ConnectionRefused

### ISSUE THREAD

**Issue by PhillyWebGuy** [paritytech/subxt#488]
I've now tried this on two different Ubuntu 20.x instances. My code works on my Mac laptop. But when executing on my Linux instance, I get this upon instantiation of the client:

`Error: Os { code: 111, kind: ConnectionRefused, message: "Connection refused" }`

My client instantiation looks like this:

`let runtime = ClientBuilder::new()
            .set_url(String::from("ws://127.0.0.1:9944"))
            .build()
            .await?
            .to_runtime_api::<my_custom_node::RuntimeApi<DefaultConfig, DefaultExtra<DefaultConfig>>>();`

Also this works:

`telnet 127.0.0.1 9944
Trying 127.0.0.1...
Connected to 127.0.0.1.
Escape character is '^]'.`

And this:

`sudo netstat -lnp | grep 9944`
`tcp        0      0 127.0.0.1:9944          0.0.0.0:*               LISTEN      22290/target/releas`

**Comment by niklasad1** [comment]
hey @PhillyWebGuy 

This is a jsonrpc related issue, you don't have any logs from jsonrpsee or your substrate node?

IIRC, we need to update subxt to use tracing for the jsonrpsee logs to show up....

**Comment by PhillyWebGuy** [comment]
Thanks for the info on how to debug. I will likely start using that with subsequent development.

But I did manage to resolve the problem late last night by starting the node with this command:

`cargo run --release -- --dev --tmp --listen-addr=/ip4/0.0.0.0/tcp/9933`

The key difference being the listen-addr I specified in my start-up command, which is all interfaces (0.0.0.0). 

I had the problem on both an Ubuntu VM and a dedicated Ubuntu server. But not on my local Mac laptop. So I'm not entirely sure what the underlying cause is that require I use 0.0.0.0 to listen on all interfaces on Ubuntu but not my Mac.
